### PR TITLE
Sd input component

### LINKED ISSUE #47: Create Input Component

### PR DESCRIPTION
**Pull request by sandaiiyahh**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

- Created `Input` reusable component using Material UI's `TextField`
- `Input` component takes in props (label, name, value, onChange, required) Thank you @isaabutaa for the amazing idea! 🤩
- Applied `Input` component to entering an existing token, filtering items, and adding a new item

## Related Issue

closes #47 

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

Screen Shot 2021-08-31 at 10 02 59 PM

Screen Shot 2021-08-31 at 10 03 34 PM

Screen Shot 2021-08-31 at 10 03 39 PM

### After

Screen Shot 2021-08-31 at 9 59 43 PM

Screen Shot 2021-08-31 at 10 00 00 PM

Screen Shot 2021-08-31 at 10 00 33 PM

## Testing Steps / QA Criteria

-  Pull down `sd-input-component`
- Checkout to `sd-input-component`
- Check to see if you are still able to enter an existing token using the new input
- Check to see if you are still able to filter properly using the new input
- Check to see if you are still able to add a new item using the new input

### CODE REVIEW

File: src/components/AddItem.js [hunk 1]
```diff
@@ -74,15 +75,13 @@ const AddItem = () => {
...
       <form onSubmit={handleClick}>
-        <label htmlFor="item">Item Name:</label>
-        <input
-          type="text"
+        <h2>Add new item</h2>
+        <Input
+          required
           name="item"
-          id="item"
+          placeholder="Item Name"
```
Comment (hulyak) [inline src/components/AddItem.js:139]: I don't know why but I cannot see the placeholders in the input fields. Does anyone have the same problem?

**Review by hulyak — COMMENTED** [review]
Good job, Sandy! I can use the input field to enter an existing token, filter items, and add an item. to the list. Just I cannot see the placeholder text. I don't know what is causing that. 
Screen Shot 2021-08-31 at 9 18 15 PM

**Comment by sandaiiyahh** [comment]
> Good job, Sandy! I can use the input field to enter an existing token, filter items, and add an item. to the list. Just I cannot see the placeholder text. I don't know what is causing that.
> Screen Shot 2021-08-31 at 9 18 15 PM

Ooh thanks for catching that Hulya! I changed it last minute and forgot to look over it. I was passing placeholder as props but forgot to change the “label” to placeholder in the Input component. Will update tomorrow!

**Review by ksiman14 — APPROVED** [review]
This looks great @sandaiiyahh! I love the styling with the green outline when you're typing in the input field. Only comment I have is the same comment as Hulya on the placeholder text 😊

**Comment by sandaiiyahh** [comment]
> This looks great @sandaiiyahh! I love the styling with the green outline when you're typing in the input field. Only comment I have is the same comment as Hulya on the placeholder text 😊

Yeah it was part of Hulya and Isa's UI theme! I was thinking we might need to change have a white background over the main part of the app though, to see everything a bit better 🧐

File: src/components/Input.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      name={name}
+      value={value}
+      variant="outlined"
+      size="small"
+      placeholder={placeholder}
+      className={classes.textField}
+      onChange={onChange}
+    />
+  );
+};
```
Comment (isaabutaa) [inline src/components/Input.js:27]: Wonderful job @sandaiiyahh! I love your use of the props and styling for this component. I checked it out on the production site, and everything looks great. 🚀

**Review by isaabutaa — APPROVED** [review]

**Pull request merged**